services/can_logger/can_logger.py

```diff
@@ -59,93 +74,220 @@ def db_init():
 
     return db
 
+# This detection function was taken from can_watchdog. Author: Aaron Neustedter
+
+def detect_can_interfaces():
+
+    can_interfaces = []
+
+    print('Gathering all can interfaces')
+
+    sysclass = '/mnt/host/sys/class/net/'
+
+    # Iterate through all links listed in /sys/class/net
+    for network in os.listdir(sysclass):
+
+        # This file defines the type of the network
+        path = sysclass + network + '/type'
+        print(f'Checking network  {network},  type at path {path}')
+
+        # Sometimes things are not setup like we expect. Live and let live
+        if not os.path.isfile(path):
+            print(f'{network} does not have a type file. Skipping')
+            continue
+
+        # Open the file and read it
+        with open(path) as typefile:
+            networktype = typefile.read().strip()
+
+        # 280 is the type for CAN. 'Documentation' here:
+        # https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/if_arp.h#L56
+        if networktype.isdigit() and int(networktype) == 280:
+            print('\t', network, ' appears to be a CAN link')
+            can_interfaces.append(network)
+
+    if len(can_interfaces) <= 0:
+        print('FATAL: No CAN interfaces found')
+        sys.exit(-1)
+
+    print(len(can_interfaces), ' found: ', can_interfaces)
+
+    return can_interfaces
+
+
+def log_can(can_interface):
+
+    print(f'Logging {can_interface}')
+    
+    frame = ''
+    rx_buff = []
+    wr_buff = []
+    socket_connected = False
+    buff = collections.deque(maxlen=3)
+    
+    # Initialize socket
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+    # Connect to socketcand
+
+    while(socket_connected is False):
+
+        try:
+
+            s.connect((host_ip, int(host_port)))
+
+        except(ConnectionRefusedError):
+
+            print('Could not connect to socketcand. Connection Refused. \
+                    Retrying...')
+            time.sleep(10)
+            socket_connected = False
+
+        else:
+
+            print(f'Successfully connected to socketcand at',
+                  '{host_ip}: {host_port}')
+            socket_connected = True
+
+    # Receive socketcand's response
+
+    s.recv(32)
+
+    # Connect to exposed CAN interface
+
+    s.sendall(b'< open ' + can_interface.encode('utf-8') + b' >')
+
+    s.recv(32)
+
+    # Set socket to 'rawmode' to receive every frame on the bus.
+    s.sendall(b'< rawmode >')
+
+    s.recv(32)
```

Comment:
As discussed in 11/5 meeting would love to see comments explaining the expected response/socketcan documentation discussing the initialization process